src/DGmethods/DGmodel_kernels.jl

```diff
@@ -31,923 +31,852 @@ const _n1, _n2, _n3 = Grids._n1, Grids._n2, Grids._n3
+      l_Q[s] = Q[ijk, s, e]
+    end
 
-          @unroll for s = 1:nauxstate
-            l_aux[s] = auxstate[ijk, s, e]
-          end
+    @unroll for s = 1:nauxstate
+      l_aux[s] = auxstate[ijk, s, e]
+    end
 
-          @unroll for s = 1:nviscstate
-            l_Qvisc[s] = Qvisc[ijk, s, e]
-          end
-          
-          @unroll for s = 1:nhyperviscstate
-            l_Qhypervisc[s] = Qhypervisc_grad[ijk, s, e]
-          end
+    @unroll for s = 1:nviscstate
+      l_Qvisc[s] = Qvisc[ijk, s, e]
+    end
+    
+    @unroll for s = 1:nhyperviscstate
+      l_Qhypervisc[s] = Qhypervisc_grad[ijk, s, e]
+    end
 
-          fill!(l_F, -zero(eltype(l_F)))
-          flux_nondiffusive!(bl, Grad{vars_state(bl,FT)}(l_F),
-                             Vars{vars_state(bl,FT)}(l_Q),
-                             Vars{vars_aux(bl,FT)}(l_aux), t)
+    fill!(l_F, -zero(eltype(l_F)))
+    flux_nondiffusive!(bl, Grad{vars_state(bl,FT)}(l_F),
+                       Vars{vars_state(bl,FT)}(l_Q),
+                       Vars{vars_aux(bl,FT)}(l_aux), t)
 
-          @unroll for s = 1:nstate
-            s_F[1,i,j,k,s] = l_F[1,s]
-            s_F[2,i,j,k,s] = l_F[2,s]
-            s_F[3,i,j,k,s] = l_F[3,s]
-          end
-          
-          fill!(l_F, -zero(eltype(l_F)))
-          flux_diffusive!(bl, Grad{vars_state(bl,FT)}(l_F),
-                          Vars{vars_state(bl,FT)}(l_Q),
-                          Vars{vars_diffusive(bl,FT)}(l_Qvisc),
-                          Vars{vars_hyperdiffusive(bl,FT)}(l_Qhypervisc),
-                          Vars{vars_aux(bl,FT)}(l_aux), t)
-
-          @unroll for s = 1:nstate
-            s_F[1,i,j,k,s] += l_F[1,s]
-            s_F[2,i,j,k,s] += l_F[2,s]
-            s_F[3,i,j,k,s] += l_F[3,s]
-          end
-          
-          # Build "inside metrics" flux
-          @unroll for s = 1:nstate
-            F1, F2, F3 = s_F[1,i,j,k,s], s_F[2,i,j,k,s], s_F[3,i,j,k,s]
-
-            s_F[1,i,j,k,s] = M * (ξ1x1 * F1 + ξ1x2 * F2 + ξ1x3 * F3)
-            if dim == 3 || (dim == 2 && direction == EveryDirection)
-              s_F[2,i,j,k,s] = M * (ξ2x1 * F1 + ξ2x2 * F2 + ξ2x3 * F3)
-            end
-            if dim == 3 && direction == EveryDirection
-              s_F[3,i,j,k,s] = M * (ξ3x1 * F1 + ξ3x2 * F2 + ξ3x3 * F3)
-            end
-          end
+    @unroll for s = 1:nstate
+      s_F[1,i,j,k,s] = l_F[1,s]
+      s_F[2,i,j,k,s] = l_F[2,s]
+      s_F[3,i,j,k,s] = l_F[3,s]
+    end
+    
+    fill!(l_F, -zero(eltype(l_F)))
+    flux_diffusive!(bl, Grad{vars_state(bl,FT)}(l_F),
+                    Vars{vars_state(bl,FT)}(l_Q),
+                    Vars{vars_diffusive(bl,FT)}(l_Qvisc),
+                    Vars{vars_hyperdiffusive(bl,FT)}(l_Qhypervisc),
+                    Vars{vars_aux(bl,FT)}(l_aux), t)
+
+    @unroll for s = 1:nstate
+      s_F[1,i,j,k,s] += l_F[1,s]
+      s_F[2,i,j,k,s] += l_F[2,s]
+      s_F[3,i,j,k,s] += l_F[3,s]
+    end
+    
+    # Build "inside metrics" flux
```

Comment:
While you're updating this, would you be able to expand on what each block is doing? e.g. the equation it is evaluating?

Reply:
Considering the size of this PR I think this should be addressed in a separate one. I don't mind doing this but I am not sure which equations you have in mind (physical equations in strong form, weak DG formulation, or fully discrete ? )

Reply:
This is probably good to do in a separate PR after I finish updating the numerics document. We can then make sure the design docs, code, and comments all agree in notation and nomenclature.